cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletUtils.java

```diff
@@ -39,4 +185,160 @@ static String getClientIpAddress(HttpServletRequest request) {
+  }
+
+  static boolean wantJSON(HttpServletRequest request) {
+    return getBooleanParam(request, JSON_PARAM, false);
+  }
+
+  static boolean allowCapacityEstimation(HttpServletRequest request) {
+    return getBooleanParam(request, ALLOW_CAPACITY_ESTIMATION_PARAM, true);
+  }
+
+  static boolean wantMaxLoad(HttpServletRequest request) {
+    return getBooleanParam(request, MAX_LOAD_PARAM, false);
+  }
+
+  static boolean isVerbose(HttpServletRequest request) {
+    return getBooleanParam(request, VERBOSE_PARAM, false);
+  }
+
+  static boolean isSuperVerbose(HttpServletRequest request) {
+    return getBooleanParam(request, SUPER_VERBOSE_PARAM, false);
+  }
+
+  static boolean clearMetrics(HttpServletRequest request) {
+    return getBooleanParam(request, CLEAR_METRICS_PARAM, true);
+  }
+
+  private static boolean getMode(HttpServletRequest request) {
+    return getBooleanParam(request, KAFKA_ASSIGNER_MODE_PARAM, false);
+  }
+
+  static boolean ignoreProposalCache(HttpServletRequest request) {
+    return getBooleanParam(request, IGNORE_PROPOSAL_CACHE_PARAM, false);
+  }
+
+  static boolean getDryRun(HttpServletRequest request) {
+    return getBooleanParam(request, DRY_RUN_PARAM, true);
+  }
+
+  static boolean throttleAddedOrRemovedBrokers(HttpServletRequest request, EndPoint endPoint) {
+    return endPoint == ADD_BROKER ? getBooleanParam(request, THROTTLE_ADDED_BROKER_PARAM, true)
+                                  : getBooleanParam(request, THROTTLE_REMOVED_BROKER_PARAM, true);
+  }
+
+  static long time(HttpServletRequest request) {
+    String timeString = request.getParameter(TIME_PARAM);
+    return (timeString == null || timeString.toUpperCase().equals("NOW")) ? System.currentTimeMillis()
+                                                                          : Long.parseLong(timeString);
+  }
+
+  static Long startMs(HttpServletRequest request) {
+    String startMsString = request.getParameter(START_MS_PARAM);
+    return startMsString == null ? null : Long.valueOf(startMsString);
+  }
+
+  static Long endMs(HttpServletRequest request) {
+    String endMsString = request.getParameter(END_MS_PARAM);
+    return endMsString == null ? null : Long.valueOf(endMsString);
+  }
+
+  static Pattern topic(HttpServletRequest request) {
+    String topicString = request.getParameter(TOPIC_PARAM);
+    return topicString != null ? Pattern.compile(topicString) : null;
+  }
+
+  static String resourceString(HttpServletRequest request) {
+    String resourceString = request.getParameter(RESOURCE_PARAM);
+    if (resourceString == null) {
+      resourceString = DEFAULT_PARTITION_LOAD_RESOURCE;
+    }
+    return resourceString;
+  }
+
+  static List<String> getGoals(HttpServletRequest request) {
+    boolean isKafkaAssignerMode = getMode(request);
+    List<String> goals;
+    String goalsString = request.getParameter(GOALS_PARAM);
+    // TODO: Handle urlencoded value (%2C instead of ,)
```

Comment:
I think we should fix this cause it'll cause a serious problem now.

Reply:
I agree that it is better to piggyback the fix for this legacy issue as part of this patch.